pkg/proxy/proxy.go

```diff
@@ -1,267 +0,0 @@
-	if len(allAttrs) == 0 {
-		msg := "Bad Request. The request or configuration is malformed."
-		klog.V(2).Info(msg)
-		http.Error(w, msg, http.StatusBadRequest)
-		return false
-	}
-
-	for _, attrs := range allAttrs {
-		// Authorize
-		authorized, reason, err := h.Authorize(ctx, attrs)
-		if err != nil {
-			msg := fmt.Sprintf("Authorization error (user=%s, verb=%s, resource=%s, subresource=%s)", u.User.GetName(), attrs.GetVerb(), attrs.GetResource(), attrs.GetSubresource())
-			klog.Errorf("%s: %s", msg, err)
-			http.Error(w, msg, http.StatusInternalServerError)
-			return false
-		}
-		if authorized != authorizer.DecisionAllow {
-			msg := fmt.Sprintf("Forbidden (user=%s, verb=%s, resource=%s, subresource=%s)", u.User.GetName(), attrs.GetVerb(), attrs.GetResource(), attrs.GetSubresource())
-			klog.V(2).Infof("%s. Reason: %q.", msg, reason)
-			http.Error(w, msg, http.StatusForbidden)
-			return false
-		}
-	}
-
-	if h.Config.Authentication.Header.Enabled {
-		// Seemingly well-known headers to tell the upstream about user's identity
-		// so that the upstream can achieve the original goal of delegating RBAC authn/authz to kube-rbac-proxy
-		headerCfg := h.Config.Authentication.Header
-		req.Header.Set(headerCfg.UserFieldName, u.User.GetName())
-		req.Header.Set(headerCfg.GroupsFieldName, strings.Join(u.User.GetGroups(), headerCfg.GroupSeparator))
-	}
-
-	return true
-}
-
-func newKubeRBACProxyAuthorizerAttributesGetter(authzConfig *authz.Config) *krpAuthorizerAttributesGetter {
-	return &krpAuthorizerAttributesGetter{authzConfig}
-}
-
-type krpAuthorizerAttributesGetter struct {
-	authzConfig *authz.Config
-}
-
-// GetRequestAttributes populates authorizer attributes for the requests to kube-rbac-proxy.
-func (n krpAuthorizerAttributesGetter) GetRequestAttributes(u user.Info, r *http.Request) []authorizer.Attributes {
-	apiVerb := ""
-	switch r.Method {
-	case "POST":
-		apiVerb = "create"
-	case "GET":
-		apiVerb = "get"
-	case "PUT":
-		apiVerb = "update"
-	case "PATCH":
-		apiVerb = "patch"
-	case "DELETE":
-		apiVerb = "delete"
-	}
-
-	var allAttrs []authorizer.Attributes
-
-	defer func() {
-		for attrs := range allAttrs {
-			klog.V(5).Infof("kube-rbac-proxy request attributes: attrs=%#+v", attrs)
-		}
-	}()
-
-	if n.authzConfig.ResourceAttributes == nil {
-		// Default attributes mirror the API attributes that would allow this access to kube-rbac-proxy
-		allAttrs := append(allAttrs, authorizer.AttributesRecord{
-			User:            u,
-			Verb:            apiVerb,
-			Namespace:       "",
-			APIGroup:        "",
-			APIVersion:      "",
-			Resource:        "",
-			Subresource:     "",
-			Name:            "",
-			ResourceRequest: false,
-			Path:            r.URL.Path,
-		})
-		return allAttrs
-	}
-
-	if n.authzConfig.Rewrites == nil {
-		allAttrs := append(allAttrs, authorizer.AttributesRecord{
-			User:            u,
-			Verb:            apiVerb,
-			Namespace:       n.authzConfig.ResourceAttributes.Namespace,
```

Comment:
Do these values match what was provided in the --config from back in main.go?

Reply:
Yes. The [configfile](https://github.com/brancz/kube-rbac-proxy/blob/master/main.go#L151-L163=) is unmarshaled into the Config type that has the [Namespace](https://github.com/brancz/kube-rbac-proxy/blob/master/pkg/authz/auth.go#L60=) `ResourceAttributes`.

Reply:
But only in the particular case of `n.authzConfig.Rewrites == nil`.